components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,423 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "authorization.h"
+#include "OauthHttpLocalServer.h"
+#include "SpiffsManger.h"
+#include "OauthAPICall.h"
+#include "cJSON.h"
+
+// ****************************** Global Variables
+OAuthInterfaceHandler_t *AuthInterfaceHandler;
+OAuthPrivateHandler_t AuthPrivateHandler;
+
+// ****************************** Local Variables
+static const char *TAG = "OAuthTask";
+
+/**
+ * @brief This function searches for specific patterns ('code' and 'state') within a character 
+ *        array and returns a boolean value indicating if either pattern was Found.
+ * @param[in] Response The character array to search within, 
+ *            and Response is response from first stage from Service athurisiation
+ * @param[in] SizeRes The size of the character array.
+ * @return Returns true if either the 'code' or 'state' pattern was Found, and false otherwise.
+ */
+static bool FindCode(char *Response, uint16_t SizeRes)
```

Comment:
please replace it in to the OauthAPICall.c

Reply:
done.